File: mobile/lib/model/buy.dart
 
import 'package:flutter/material.dart';

import 'package:http/http.dart' as http;
class Buy{
  final String id;
  final String title;
  final String owner;
  final String description;
  final String image;
  final int size;
  final int room_number;
  final String location;
  final num price;
  const Buy({
    required this.id,
    required this.title,
    required this.owner,
    required this.description,
    required this.image,
    required this.size,
    required this.room_number,
    required this.location,
    required this.price
  });
  factory Buy.fromJson(Map<String, dynamic> json) {
    return Buy(
        id: json['id'],
        title: json['title'],
        owner: json['owner'],
        description: json['description'],
        image: json['image'],
        size: json['size'],
        room_number: json['room_number'],
        location: json['location'],
        price: json['price']
    );
  }
}

File: Backend/models/buy.js
const mongoose = require('mongoose');

const buySchema = new mongoose.Schema({
  title: {type: string, required: true},
  
  owner: {type: string, required: true},
  description: {type: string, required: true},
  image: {type: string, required: true},
  
  size: {type: number, required: true},
  
  room_number: {type: number, required: true},
  location: {type: string, required: true},
  price: {type: number, required: true}
});

module.exports = mongoose.model('Buy', buySchema);